imgr.py

```diff
@@ -42,6 +42,26 @@ def usage():
     print "Also, --add, PORT may also be specified as 'open' or 'closed'."
     sys.exit(1)
 
+# Convert the port string to an integer.
+# We need the host and the config list in case of 'open' or 'closed'.
+def port_to_int(port, host, cl):
+    if port != 'closed' and port != 'open':
+        return int(port)
+    plist = []
+    for c in cl:
+        if c['host'] == host:
+            plist.append(int(c['port']))
+    if port == 'closed':
+        r = range(30001, 39000)
+    else:
+        r = range(39100, 39200)
+    for i in r:
+        if i not in plist:
+            print "Choosing %s port %d" % (port, i)
+            return i
+    return None  # We should never really get here.
```

Comment:
If we don't want to ever get here we could just `raise ValueError(f'invalid port supplied {port}')`, the rest of the program will probably throw eventually.  Throwing here probably makes things easier to track down

Reply:
The main way we get here is when we run out of ports, right?

Reply:
Point taken.  I will modify this on Monday.

Reply:
Slightly different message, but raising an error instead of returning None now.

Reply:
We still have an afs repo.  This doesn't seem to be in sync with github at all... it was missing Josh's last two commits from his PR.  I had accidentally pushed there, and thought I was pushing here.

I cherry-picked the last two from Josh, so afs *is* now in sync with what this will be after the PR closes.

We should really kill the afs repo though.

Reply:
😬 sorry I should have pushed it to afs!